host/sgx/enter.c

```diff
@@ -44,14 +45,18 @@
 #endif
 
 // The following registers are inputs to ENCLU instruction. They are also
-// clobbered. Hence marked as +r.
+// clobbered and hence are marked as +r.
 #define OE_ENCLU_REGISTERS \
     "+r"(rax), "+r"(rbx), "+r"(rcx), "+r"(rdi), "+r"(rsi), "+r"(rdx)
 
 // The following registers are clobbered by ENCLU.
-// Only rbp and rsp are preserved.
-#define OE_ENCLU_CLOBBERED_REGISTERS \
-    "r8", "r9", "r10", "r11", "r12", "r13", "r14", "r15"
+// Only rbp and rsp are preserved on return from ENCLU.
+// The XMM registers are listed as clobbered to signal to the compiler that
+// they need to be preserved when --target=x86_64-msvc-windows and are
+// ignored on Linux builds.
+#define OE_ENCLU_CLOBBERED_REGISTERS                                      \
+    "r8", "r9", "r10", "r11", "r12", "r13", "r14", "r15", "xmm6", "xmm7", \
+        "xmm8", "xmm9", "xmm10", "xmm11", "xmm12", "xmm13", "xmm14", "xmm15"
```

Comment:
Should we conservatively list "ymm" registers instead of "xmm" registers?

Reply:
No need, the Windows x64 ABI specifically states that:

> The x64 ABI considers the registers RAX, RCX, RDX, R8, R9, R10, R11, and XMM0-XMM5 volatile. When present, the upper portions of YMM0-YMM15 and ZMM0-ZMM15 are also volatile. On AVX512VL, the ZMM, YMM, and XMM registers 16-31 are also volatile.

This is also reflected in the standard prolog/epilog generated by the compiler on Windows.